src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface TodoModalProps {
+  selectedTodo: Todo | undefined,
+  selectedTodoId: number | 0,
+  setSelectedTodoId: (selectedTodoId: number | 0) => void,
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  selectedTodo,
+  selectedTodoId,
+  setSelectedTodoId,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const fetchUser = async () => {
+    const userFromServer = await getUser(selectedTodoId);
+
+    setUser(userFromServer);
+  };
+
+  fetchUser();
```

Comment:
same here about infinite cycle
check how it works
for now you call the function for fetching user, this function updates the state, `TodoModal` component renders one more time, and this function will be called again